sharding/nodesSetup_test.go

```diff
@@ -28,102 +29,157 @@ var (
 	}
 )
 
-func createAndAssignNodes(ns NodesSetup, noOfInitialNodes int) *NodesSetup {
-	ns.InitialNodes = make([]*InitialNode, noOfInitialNodes)
+func createAndAssignNodes(ns sharding.NodesSetup, noOfInitialNodes int) *sharding.NodesSetup {
+	ns.InitialNodes = make([]*sharding.InitialNode, noOfInitialNodes)
 
 	for i := 0; i < noOfInitialNodes; i++ {
 		lookupIndex := i % len(PubKeys)
-		ns.InitialNodes[i] = &InitialNode{}
+		ns.InitialNodes[i] = &sharding.InitialNode{}
 		ns.InitialNodes[i].PubKey = PubKeys[lookupIndex]
 		ns.InitialNodes[i].Address = Address[lookupIndex]
 	}
 
-	err := ns.processConfig()
+	err := ns.ProcessConfig()
 	if err != nil {
 		return nil
 	}
 
-	ns.processMetaChainAssigment()
-	ns.processShardAssignment()
-	ns.createInitialNodesInfo()
+	ns.ProcessMetaChainAssigment()
+	ns.ProcessShardAssignment()
+	ns.CreateInitialNodesInfo()
 
 	return &ns
 }
 
-func createNodesSetupOneShardOneNodeWithOneMeta() *NodesSetup {
+func createNodesSetupOneShardOneNodeWithOneMeta() *sharding.NodesSetup {
 	noOfInitialNodes := 2
-	ns := &NodesSetup{}
+	ns := &sharding.NodesSetup{}
 	ns.ConsensusGroupSize = 1
 	ns.MinNodesPerShard = 1
 	ns.MetaChainConsensusGroupSize = 1
 	ns.MetaChainMinNodes = 1
+	ns.InitialNodes = make([]*sharding.InitialNode, noOfInitialNodes)
+	ns.InitialNodes[0] = &sharding.InitialNode{}
+	ns.InitialNodes[0].PubKey = PubKeys[0]
+	ns.InitialNodes[0].Address = Address[0]
+	ns.InitialNodes[1] = &sharding.InitialNode{}
+	ns.InitialNodes[1].PubKey = PubKeys[1]
+	ns.InitialNodes[1].Address = Address[1]
+	err := ns.ProcessConfig()
+	if err != nil {
+		return nil
+	}
 
-	return createAndAssignNodes(*ns, noOfInitialNodes)
+	ns.ProcessMetaChainAssigment()
+	ns.ProcessShardAssignment()
+	ns.CreateInitialNodesInfo()
+
+	return ns
 }
 
-func createNodesSetupTwoShardTwoNodesWithOneMeta() *NodesSetup {
+func createNodesSetupTwoShardTwoNodesWithOneMeta() *sharding.NodesSetup {
 	noOfInitialNodes := 6
-	ns := &NodesSetup{}
+	ns := &sharding.NodesSetup{}
 	ns.ConsensusGroupSize = 1
 	ns.MinNodesPerShard = 2
 	ns.MetaChainConsensusGroupSize = 1
 	ns.MetaChainMinNodes = 2
+	ns.InitialNodes = make([]*sharding.InitialNode, noOfInitialNodes)
 
-	return createAndAssignNodes(*ns, noOfInitialNodes)
+	for i := 0; i < noOfInitialNodes; i++ {
```

Comment:
Duplicated code here, and 2 functions below

Reply:
the configurations are slighly different.